hardware/SensorPi/main.py

```diff
@@ -12,6 +14,20 @@
 sense.show_message("MERCURY", text_colour=nyu_purple, scroll_speed=0.04)
 sense.clear()
 
+# initilaize time of last sent temperature
+fmt = "%Y-%m-%d %H:%M:%S"
+time_of_last_sent_temperature = datetime.now().strftime(fmt)
+time_of_last_sent_pressure = datetime.now().strftime(fmt)
+time_of_last_sent_humidity = datetime.now().strftime(fmt)
+time_of_last_sent_acceleration = datetime.now().strftime(fmt)
+time_of_last_sent_orientation = datetime.now().strftime(fmt)
```

Comment:
this is super minor, but maybe it would be better to do a single call to datetime.now instead of 5. e.g. val1 = val2 = val4 = datetime.now.

This is a nit, but might save a tiny bit of processing.

Reply:
Yup, agreed.